src/ring_theory/polynomial/cyclotomic/basic.lean

```diff
@@ -423,6 +423,31 @@ begin
            geom_sum_mul, prod_cyclotomic_eq_X_pow_sub_one h]
 end
 
+lemma cyclotomic_dvd_geom_sum_of_dvd (R) [comm_ring R] [is_domain R] {d n : ℕ} (hdn : d ∣ n)
```

Comment:
Since both `cyclotomic` and `geom_sum` commutes with any base change, the fact that this holds over `Z` implies that it holds for any `comm_ring` (so `is_domain` shouldn't be needed).

Reply:
hmm, what do you have in mind? because `map_dvd_map` requires that `cast_ring_hom` is injective, which instead requires char-zero:

```lean
lemma cyclotomic_dvd_geom_sum_of_dvd (R) [comm_ring R] [char_zero R] {d n : ℕ} (hdn : d ∣ n)
  (hd : d ≠ 1) : cyclotomic d R ∣ geom_sum X n :=
begin
  suffices : cyclotomic d ℤ ∣ geom_sum X n,
  { have key := (map_ring_hom (int.cast_ring_hom R)).map_geom_sum X n,
    simp only [coe_map_ring_hom, map_X] at key,
    rwa [←map_dvd_map (int.cast_ring_hom R), key, map_cyclotomic] at this,
    exact int.cast_injective,
    exact cyclotomic.monic _ _ },
  rcases n.eq_zero_or_pos with rfl | hn,
  { simp },
  rw ←prod_cyclotomic_eq_geom_sum hn,
  apply finset.dvd_prod_of_mem,
  simp [hd, hdn, hn.ne']
end
```

maybe I should go get lunch and then think about this more

Reply:
Write `geom_sum X n = P * cyclotomic d ℤ`, take `map` on both sides and you get `geom_sum X n = P.map _ * cyclotomic d R`. Once implication of `map_dvd_map` (the one you need here) shouldn't have any condition on `f`.

Reply:
ah, yes, of course; and this is juust `eval_2_dvd`, but it hadn't been specialized. Thaks!